test/vfd_swmr_group_writer.c

```diff
@@ -440,123 +2928,90 @@ main(int argc, char **argv)
         goto error;
     }
 
+    /* Pass the named pipe information to the struct of state_t s, for attribute tests.*/
+    if(s.use_named_pipes) {
+        s.np_fd_w_to_r = fd_writer_to_reader;
+        s.np_fd_r_to_w = fd_reader_to_writer;
+        s.np_notify = notify;
+        s.np_verify = verify;
+        s.tick_len  = config.tick_len;
+        s.max_lag   = config.max_lag;
+    }
+
+    /* For attribute test, force the named pipe to communicate in every step. */
```

Comment:
For attribute testing, may I know why the synchronization has to be very step?

Reply:
For each operation(either group or attribute),  I want the hand-shake between to the reader and the writer. I just want to make sure that there is no reader-go-ahead issue like the general group test .

Reply:
But is it possible for the user to choose "every step" or "every n steps"?

Reply:
No, not for attribute tests when named pipe is used. Since in every step, a group is created. This is one operation.  As long as this exists, a possible reader ahead issue exists and a fake reader verification failure error may happen.  That's why Ray uses 5 ticks(instead of the default 3) for readers to wait in the default group test.  Users can choose every n steps for the default group test. This is the original test that has an attribute added for each group.  Users can use that option to add attributes for the group. However, if you think without using the named pipe, it benefits the users with all the attribute tests for uses to choose either every n steps or every 1 step. I can change the code. Without using the named pipe, there is no reason to force the attribute operations in every step.

Reply:
Thank you for the explanation.